template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java

```diff
@@ -49,6 +51,11 @@ protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, Templat
                             config(DFS_REPLICATION, DFS_REPLICATION_VALUE.toString())
                     );
                 }
+                if (entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId())) {
```

Comment:
The previous block is also about `isSDXOptimizationEnabled`. This block check that entitlement but is also applied for Datahubs, this is confusing. I would clean up the previous method name seems that looks more specific.

Reply:
https://jira.cloudera.com/browse/CDPSDX-3528?focusedCommentId=5522462&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-5522462

Reply:
@RokolyA94 This change is limited to only SDX. Please update the code to add the check.

Reply:
stack type check added.
restricted only for DL.